x/gov/types/keys.go

```diff
@@ -41,6 +41,7 @@ var (
 	ActiveProposalQueuePrefix   = []byte{0x01}
 	InactiveProposalQueuePrefix = []byte{0x02}
 	ProposalIDKey               = []byte{0x03}
+	ProposalContentsKeyPrefix   = []byte{0x04}
```

Comment:
There's an alternative design where we don't change the existing state (no need to set messages=nil and metadata=nil on existing proposals).

We do:
```suggestion
	VotingPeriodProposalsKeyPrefix   = []byte{0x04}
```
which contains `0x04 | <proposal_id> => 0x01` for proposals still in voting period.

MsgVote just needs to know if a proposal is in voting period or not. Currently, we unmarshal the whole proposal struct to get this. What we can do is create a new key that only holds the proposals in voting period.

Reply:
This makes a lot of sense. There's no other use case (that I know of) for having the messages and metadata separated so far. So this is a better approach imo. wdyt @tac0turtle ?

Reply:
yea that is fine, its a bit of a larger change but make sense